crates/extensions/c8y_mapper_ext/src/json.rs

```diff
@@ -36,18 +36,20 @@ pub enum CumulocityJsonError {
 pub fn from_thin_edge_json(
     input: &str,
     entity: &EntityMetadata,
+    m_type: &str,
 ) -> Result<String, CumulocityJsonError> {
     let timestamp = WallClock.now();
-    let c8y_vec = from_thin_edge_json_with_timestamp(input, timestamp, entity)?;
+    let c8y_vec = from_thin_edge_json_with_timestamp(input, timestamp, entity, m_type)?;
     Ok(c8y_vec)
 }
 
 fn from_thin_edge_json_with_timestamp(
     input: &str,
     timestamp: OffsetDateTime,
     entity: &EntityMetadata,
+    m_type: &str,
 ) -> Result<String, CumulocityJsonError> {
-    let mut serializer = serializer::C8yJsonSerializer::new(timestamp, entity);
+    let mut serializer = serializer::C8yJsonSerializer::new(timestamp, entity).with_type(m_type);
```

Comment:
I would simply add an extra argument to the constructor as this is the only place where a `C8yJsonSerializer` is built.

```suggestion
    let mut serializer = serializer::C8yJsonSerializer::new(timestamp, entity, m_type);
```